File: chapter05/src/main/scala/Car.scala

class Car(val manufacturer: String, val modelName: String, val modelYear: Int = -1, var licensePlate: String = "") {

  override def toString = s"Car($manufacturer, $modelName, $modelYear, $licensePlate)"
}

//Test
println(new Car("Honda", "Civic", 2011, "xx123zz"))
println(new Car("Hummer", "H1", 2010))
println(new Car("Opel", "Astra"))

File: chapter05/src/main/java/Car.java

public class Car {

    private String manufacturer;
    private String modelName;
    private int modelYear;
    private String licensePlate;

    public Car(String manufacturer, String modelName, int modelYear, String licensePlate) {
        initialize(manufacturer, modelName, modelYear, licensePlate);
    }

    public Car(String manufacturer, String modelName) {
        initialize(manufacturer, modelName, -1, "");
    }

    public Car(String manufacturer, String modelName, int modelYear) {
        initialize(manufacturer, modelName, modelYear, "");
    }

    public Car(String manufacturer, String modelName, String licensePlate) {
        initialize(manufacturer, modelName, -1, licensePlate);
    }

    private void initialize(String manufacturer, String modelName, int modelYear, String licensePlate) {
        this.manufacturer = manufacturer;
        this.modelName = modelName;
        this.modelYear = modelYear;
        this.licensePlate = licensePlate;
    }

    public String getManufacturer() {
        return manufacturer;
    }

    public String getModelName() {
        return modelName;
    }

    public int getModelYear() {
        return modelYear;
    }

    public String getLicensePlate() {
        return licensePlate;
    }

    public void setLicensePlate(String licensePlate) {
        this.licensePlate = licensePlate;
    }
}
